[Bug]: build racket failed ndk25

### Problem description

make[4]: *** [Makefile:212: xsrc/bool.c] Error 1
make[4]: *** Waiting for unfinished jobs....
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:224: xsrc/env.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:210: xsrc/bignum.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:220: xsrc/complex.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:216: xsrc/compenv.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:222: xsrc/dynext.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:228: xsrc/eval.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:208: xsrc/salloc.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
make[4]: *** [Makefile:226: xsrc/error.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
Waiting on lock: /home/builder/.termux-build/racket/host-build/racket/gc2/../../compiled/SETUP-LOCKmake[4]: *** [Makefile:214: xsrc/char.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
Waiting on lock: /home/builder/.termux-build/racket/host-build/racket/gc2/../../compiled/SETUP-LOCKmake[4]: *** [Makefile:218: xsrc/compile.c] Error 1
precompiled-header: line mismatch with precompiled: #"#pragma GCC diagnostic push" (line 982) versus #"   return (void *) __p;"
  context...:
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:508:10: loop
   /home/builder/.termux-build/racket/src/collects/compiler/private/xform.rkt:8:2: xform
   "/home/builder/.termux-build/racket/src/src/racket/gc2/xform-mod.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   "/home/builder/.termux-build/racket/src/collects/setup/main.rkt": [running body]
   temp37_0
   for-loop
   run-module-instance!125
   perform-require!78
Waiting on lock: /home/builder/.termux-build/racket/host-build/racket/gc2/../../compiled/SETUP-LOCKmake[4]: *** [Makefile:230: xsrc/file.c] Error 1
make[4]: Leaving directory '/home/builder/.termux-build/racket/host-build/racket/gc2'
make[3]: *** [Makefile:178: all] Error 2
make[3]: Leaving directory '/home/builder/.termux-build/racket/host-build/racket/gc2'
make[2]: *** [Makefile:109: 3m] Error 2
make[2]: Leaving directory '/home/builder/.termux-build/racket/host-build/racket'
make[1]: *** [Makefile:61: 3m] Error 2

make[1]: Leaving directory '/home/builder/.termux-build/racket/host-build'
make: *** [Makefile:51: all] Error 2

[racket.txt](https://github.com/termux/termux-packages/files/9333775/racket.txt)



### What steps will reproduce the bug?

build

### What is the expected behavior?

_No response_

### System information

termux-info:

```

```

--- a/packages/racket/collects-compiler-private-xform.rkt.patch.beforehostbuild
+++ b/packages/racket/collects-compiler-private-xform.rkt.patch.beforehostbuild
@@ -1,0 +1,11 @@
+--- a/collects/compiler/private/xform.rkt
++++ b/collects/compiler/private/xform.rkt
+@@ -506,7 +506,7 @@
+         (when recorded-cpp-in
+           ;; Skip over common part:
+           (let loop ([lpos 1])
+-            (let ([pl (read-bytes-line recorded-cpp-in 'any)])
++            (let ([pl (skip-to-interesting-line recorded-cpp-in)])
+               (unless (eof-object? pl)
+                 (let ([l (skip-to-interesting-line (car cpp-process))])
+                   (unless (equal? pl l)

